### PR TITLE
fix #11 - make runnable on Heroku

### LINKED ISSUE #11: Make webalyzer runn-able on Heroku

### PR DESCRIPTION
**Pull request by groovecoder**
See live app at https://webalyzer.herokuapp.com/

### CODE REVIEW

**Comment by groovecoder** [comment]
Need to find a way to serve the collect.js static resource with CORS headers, since staticfiles seems to ignore CorsMiddleware ?

**Comment by groovecoder** [comment]
Maybe there's a wsgi-level CORS library?

**Comment by jezdez** [comment]
@groovecoder Whitenoise does the CORS already, no need for the additional app. See its source please.

File: webalyzer/base/templates/home.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{% extends "base.html" %}
+
+{% block body %}
+<h1>Webalyzer</h1>
+<h2>Collector Instructions</h2>
+<ol>
+    <li>Add the following to your page(s):
+<code>
+    &lt;script src="{{ request.scheme }}://{{ request.get_host }}/static/collect.js"
```
Comment (jezdez) [inline webalyzer/base/templates/home.html]: Why not use `{{ request.build_absolute_uri }}{% static "collect.js" %}`?

Needs a `{% load staticfiles %}` above.

**Comment by groovecoder** [comment]
Apparently whitenoise should be pre-configured for CORS, but it's not sending the CORS headers at https://webalyzer.herokuapp.com//static/collect.72e1b06b09c6.js or https://webalyzer.herokuapp.com//static/collect.js. :frowning:

**Comment by jezdez** [comment]
@groovecoder This worked for me

```
<Pal:jezdez> 10:45:17 ~ $ http -h http://webalyzer.herokuapp.com/static/collect.js
HTTP/1.1 200 OK
Access-Control-Allow-Origin: *
Cache-Control: public, max-age=60
Connection: keep-alive
Content-Length: 4575
Content-Type: application/javascript; charset="utf-8"
Date: Wed, 27 May 2015 08:45:22 GMT
Last-Modified: Tue, 26 May 2015 20:03:25 GMT
Server: gunicorn/19.3.0
Via: 1.1 vegur
```

**Comment by groovecoder** [comment]
I got back into this today. It's all running on Heroku at webalyzer.herokuapp.com, except that the gator worker process is having trouble connecting to the redis addon I added. :frowning:

@peterbe - I did `heroku config:set ALLIGATOR_CONN=redis://<url-from-addon>`. Should that be it? The error shows:

```
2015-06-19T02:32:42.969002+00:00 heroku[worker.1]: Starting process with command `python manage.py run-gator`
2015-06-19T02:32:43.644269+00:00 heroku[worker.1]: State changed from starting to up
2015-06-19T02:32:49.210005+00:00 app[worker.1]:   File "manage.py", line 10, in <module>
2015-06-19T02:32:49.210091+00:00 app[worker.1]:     execute_from_command_line(sys.argv)
2015-06-19T02:32:49.210158+00:00 app[worker.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/django/core/management/__init__.py", line 338, in execute_from_command_line
2015-06-19T02:32:49.210308+00:00 app[worker.1]:     utility.execute()
2015-06-19T02:32:49.210347+00:00 app[worker.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/django/core/management/__init__.py", line 330, in execute
2015-06-19T02:32:49.210887+00:00 app[worker.1]:     self.fetch_command(subcommand).run_from_argv(self.argv)
2015-06-19T02:32:49.210930+00:00 app[worker.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/django/core/management/base.py", line 390, in run_from_argv
2015-06-19T02:32:49.211064+00:00 app[worker.1]:     self.execute(*args, **cmd_options)
2015-06-19T02:32:49.211102+00:00 app[worker.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/django/core/management/base.py", line 441, in execute
2015-06-19T02:32:49.211236+00:00 app[worker.1]:     output = self.handle(*args, **options)
2015-06-19T02:32:49.198916+00:00 app[worker.1]: Alligator Worker (#3) starting & consuming "all".
2015-06-19T02:32:49.198977+00:00 app[worker.1]: Alligator Worker (#3) will never die.
2015-06-19T02:32:49.208632+00:00 app[worker.1]: Traceback (most recent call last):
2015-06-19T02:32:49.211272+00:00 app[worker.1]:   File "/app/webalyzer/base/management/commands/run-gator.py", line 14, in handle
2015-06-19T02:32:49.211338+00:00 app[worker.1]:     worker.run_forever()
2015-06-19T02:32:49.211373+00:00 app[worker.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/alligator/workers.py", line 98, in run_forever
2015-06-19T02:32:49.211447+00:00 app[worker.1]:     if self.gator.len():
2015-06-19T02:32:49.211482+00:00 app[worker.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/alligator/gator.py", line 78, in len
2015-06-19T02:32:49.211553+00:00 app[worker.1]:     return self.backend.len(self.queue_name)
2015-06-19T02:32:49.211598+00:00 app[worker.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/alligator/backends/redis_backend.py", line 48, in len
2015-06-19T02:32:49.211672+00:00 app[worker.1]:     return self.conn.llen(queue_name)
2015-06-19T02:32:49.211708+00:00 app[worker.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/redis/client.py", line 1202, in llen
2015-06-19T02:32:49.211992+00:00 app[worker.1]:     return self.execute_command('LLEN', name)
2015-06-19T02:32:49.212037+00:00 app[worker.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/redis/client.py", line 564, in execute_command
2015-06-19T02:32:49.212182+00:00 app[worker.1]:     connection.send_command(*args)
2015-06-19T02:32:49.212218+00:00 app[worker.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/redis/connection.py", line 556, in send_command
2015-06-19T02:32:49.212362+00:00 app[worker.1]:     self.send_packed_command(self.pack_command(*args))
2015-06-19T02:32:49.212402+00:00 app[worker.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/redis/connection.py", line 532, in send_packed_command
2015-06-19T02:32:49.212540+00:00 app[worker.1]:     self.connect()
2015-06-19T02:32:49.212576+00:00 app[worker.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/redis/connection.py", line 440, in connect
2015-06-19T02:32:49.212701+00:00 app[worker.1]:     self.on_connect()
2015-06-19T02:32:49.212735+00:00 app[worker.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/redis/connection.py", line 514, in on_connect
2015-06-19T02:32:49.212870+00:00 app[worker.1]:     if nativestr(self.read_response()) != 'OK':
2015-06-19T02:32:49.212905+00:00 app[worker.1]:   File "/app/.heroku/python/lib/python2.7/site-packages/redis/connection.py", line 574, in read_response
2015-06-19T02:32:49.213048+00:00 app[worker.1]:     raise response
2015-06-19T02:32:49.213162+00:00 app[worker.1]: redis.exceptions.ResponseError: NOAUTH Authentication required.
2015-06-19T02:32:50.238186+00:00 heroku[worker.1]: Process exited with status 1
2015-06-19T02:32:50.263188+00:00 heroku[worker.1]: State changed from up to crashed
2015-06-19T02:32:50.264128+00:00 heroku[worker.1]: State changed from crashed to starting
```

**Comment by peterbe** [comment]
Does your Redis URL contain any credentials?

**Comment by groovecoder** [comment]
Rebased on master to get celery code.

**Comment by Elchi3** [comment]
Repository is unmaintained. Closing this now.

**Pull request closed without merging**